rocksdb backend not working

I have written some very straightforward bleve code to index and search data. It works great with the default boltdb backend.

Today I tried to switch to the rocksdb backend, but it's not working. I am unable to retrieve any data out of the database. All search requests return 0 results, and DocCount() returns 0 as well.

I have confirmed rocksdb is being used correctly. I also see the database size growing on disk after I index documents:

```
$ cat index_meta.json
{"storage":"rocksdb","index_type":"upside_down"}

$ du -sh store
 36M	store

$ ls -1 store/*.sst | wc -l
      19
```

I installed rocksdb using `brew`:

```
/usr/local/Cellar/rocksdb/4.5.1/lib/librocksdb.4.5.1.dylib
```

and am using `blevex/rocksdb` as follows:

``` go
val, err = bleve.NewUsing(path, indexMapping(), bleve.Config.DefaultIndexType, rocksdb.Name, nil)
```

Was unable to build latest blevex against 3.11.2:

```
Undefined symbols for architecture x86_64:
  "_rocksdb_writebatch_deletev", referenced from:
      _blevex_rocksdb_execute_direct_batch in batchex.cgo2.o
  "_rocksdb_writebatch_mergev", referenced from:
      _blevex_rocksdb_execute_direct_batch in batchex.cgo2.o
  "_rocksdb_writebatch_putv", referenced from:
      _blevex_rocksdb_execute_direct_batch in batchex.cgo2.o
ld: symbol(s) not found for architecture x86_64
```

Looks like the batchex code is new since then, so I suspect that's where the bug is. The corrupted data in the index looks a lot like a buffer over-read, and there's a lot of pointer manipulation happening in the BatchEx implementation that might be related.

--- a/rocksdb/writer.go
+++ b/rocksdb/writer.go
@@ -29,8 +29,11 @@ func (w *Writer) NewBatch() store.KVBatch {
 }
 
 func (w *Writer) NewBatchEx(options store.KVBatchOptions) ([]byte, store.KVBatch, error) {
-	rv := newBatchEx(options)
-	return rv.buf, rv, nil
+	// Disabled due to https://github.com/blevesearch/blevex/issues/22
+	// rv := newBatchEx(options)
+	// return rv.buf, rv, nil
+
+	return make([]byte, options.TotalBytes), w.NewBatch(), nil
 }
 
 func (w *Writer) ExecuteBatch(b store.KVBatch) error {

